src/python/pants/backend/python/tasks2/python_run.py

```diff
@@ -0,0 +1,80 @@
+# coding=utf-8
+# Copyright 2016 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import signal
+
+from pex.interpreter import PythonInterpreter
+from pex.pex import PEX
+from pex.pex_builder import PEXBuilder
+
+from pants.backend.python.targets.python_binary import PythonBinary
+from pants.backend.python.tasks2.gather_sources import GatherSources
+from pants.backend.python.tasks2.resolve_requirements import ResolveRequirements
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.task.task import Task
+from pants.util.contextutil import temporary_dir
+from pants.util.strutil import safe_shlex_split
+
+
+class PythonRun(Task):
+
+  @classmethod
+  def register_options(cls, register):
+    super(PythonRun, cls).register_options(register)
+    register('--args', type=list, help='Run with these extra args to main().')
+
+  @classmethod
+  def supports_passthru_args(cls):
+    return True
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    round_manager.require_data(PythonInterpreter)
+    round_manager.require_data(ResolveRequirements.REQUIREMENTS_PEX)
+    round_manager.require_data(GatherSources.PYTHON_SOURCES)
+
+  def execute(self):
+    binary = self.require_single_root_target()
+    if isinstance(binary, PythonBinary):
+      # We can't throw if binary isn't a PythonBinary, because perhaps we were called on a
+      # jvm_binary, in which case we have to no-op and let jvm_run do its thing.
+      # TODO(benjy): Some more elegant way to coordinate how tasks claim targets.
```

Comment:
This can be accomplished with `MutexTask`, which is used for a similar case with the `repl` tasks: https://github.com/pantsbuild/pants/blob/master/src/python/pants/task/mutex_task_mixin.py